client.go

```diff
@@ -156,11 +156,11 @@ func NewSnowthClient(discover bool, addrs ...string) (*SnowthClient, error) {
 		return nil, err
 	}
 
-	return NewClient(cfg)
+	return NewClient(context.Background(), cfg)
```

Comment:
I might advise using [context.WithCancel](https://pkg.go.dev/context#WithCancel) and adding a field to the SnowthClient type for the ctx cancel function, though I need to read more of the code to determine if that would be worthwhile.

Reply:
That's not what this is being used for. Though I can see why it looks that way. This is not a context being passed in that is used for the client, and would therefore have a cancel function saved as a field on the client. This is just the context to use when running the NewClient() function. That function performs several I/O operations and needs to be able to respond to context terminations. The context would be provided by the consumer of the library. They would control whether or not it has cancelation or deadlines. This use of context.Background() is only in the NewSnowthClient() function, which only exists for backwards compatibility anyway, and is probably used nowhere outside of gosnowth's own tests. In this particular case, I think it would be ok to use a context with no cancel or timeout.

Reply:
That makes sense, thanks for the explanation!
In synopsis, you're saying that the context spawns to facilitate the creation of 
a SnowthClient, and is guaranteed to have ended by the time that the instance is
returned to the caller.

Reply:
Yes. In the event NewSnowthClient() ever gets used anywhere, that would be the behavior. But, Anywhere that uses gosnowth now would use NewClient(ctx) instead and pass in their own context. There is a lot of stuff in gosnowth to preserve backwards compatibility, and it was originally written before context.Context existed.

Reply:
That clears it up, thank you!